protocols/connection-limit/tests/lib.rs

```diff
@@ -0,0 +1,198 @@
+// TODO: Should imports go through libp2p crate?
+use futures::executor::LocalPool;
+use futures::future;
+use futures::task::LocalSpawn;
+use futures::task::LocalSpawnExt;
+use futures::task::Spawn;
+use futures::FutureExt;
+use futures::StreamExt;
+use futures::TryStreamExt;
+use libp2p::NetworkBehaviour;
+use libp2p_connection_limit::ConnectionLimits;
+use libp2p_core::connection::{ConnectionId, Endpoint};
+use libp2p_core::identity;
+use libp2p_core::identity::PublicKey;
+use libp2p_core::muxing::StreamMuxerBox;
+use libp2p_core::transport::Boxed;
+use libp2p_core::transport::TransportEvent;
+use libp2p_core::transport::{MemoryTransport, Transport};
+use libp2p_core::upgrade;
+use libp2p_core::Multiaddr;
+use libp2p_core::PeerId;
+use libp2p_plaintext::PlainText2Config;
+use libp2p_swarm::dial_opts::DialOpts;
+use libp2p_swarm::DialError;
+use libp2p_swarm::DummyBehaviour;
+use libp2p_swarm::KeepAlive;
+use libp2p_swarm::NetworkBehaviour;
+use libp2p_swarm::Swarm;
+use libp2p_swarm::SwarmBuilder;
+use libp2p_swarm::SwarmEvent;
+use libp2p_yamux::YamuxConfig;
+use quickcheck::QuickCheck;
+
+// TODO: Test that pending limit is decreased.
+
+#[test]
+fn enforces_pending_outbound_connection_limit() {
+    fn prop(outbound: u8) {
```

Comment:
Sorry for OT but we have so many different ways of testing swarms in our repo, it makes me uneasy every time :sweat_smile:

Reply:
I'll put this on my list to work on from September forward :D